perception_dataset/deepen/non_annotated_t4_to_deepen_converter.py

```diff
@@ -101,6 +103,17 @@ def _convert_one_frame(self, input_dir: str, output_dir: str, frame_index: int):
             camera_channel = camera_sensor_type.value["channel"]
             camera_token: str | None = self._get_camera_token(camera_channel, sample, nusc)
 
+            if camera_token is None:
+                if self._drop_camera_token_not_found:
+                    logger.warning(
+                        f"Skipping.. Camera token not found for {camera_channel} in frame {frame_index}. Dropping this frame"
+                    )
+                    return
+                else:
+                    logger.error(
+                        f"Camera token not found for {camera_channel} in frame {frame_index}. "
+                    )
+                    continue
```

Comment:
It looks reasonable for me that we continue the process if it fails to look up camera token and `drop_camera_token_not_found=True`, otherwise we raise a particular error.

Isn't this your intended behavior?

```suggestion
            if camera_token is None:
                if self._drop_camera_token_not_found:
                    logger.warning(
                        f"Skipping.. Camera token not found for {camera_channel} in frame {frame_index}. Dropping this frame"
                    )
                    continue
                else:
                    raise LookupError(f"Camera token not found for {camera_channel} in frame {frame_index}. "
```

Reply:
There is some difference
In case, we do not find the camera token, the user might want to either 
1. drop the entire frame ( `_drop_camera_token_not_found = True` )
2. keep the frame without the missing images ( `_drop_camera_token_not_found = False` )
3. do not process the dataset (by raising an error? )

if we use your proposed changes, we only achieve 2 and 3 but not 1.

With the current PR, we can achieve 1 and 2 by setting _drop_camera_token_not_found . And, the user can choose not to use the processed dataset if they get the f"Camera token not found for {camera_channel} in frame {frame_index}. " message , this serves same purpose as 3.

Reply:
@SamratThapa120 Ah, make sense. I misunderstood the variable as **drop camera (not frame) data if its token not found**.